dotnet/Simple.Sauce/SauceSession.cs

```diff
@@ -1,100 +1,56 @@
-﻿using System;
-using System.Collections.Generic;
-using OpenQA.Selenium;
-using OpenQA.Selenium.Chrome;
+﻿using OpenQA.Selenium;
 
 namespace Simple.Sauce
 {
     public class SauceSession
     {
-        public SauceSession()
-        {
-            Driver = new SauceDriver();
-            Options = new SauceOptions();
-        }
+        private IWebDriver _driver;
 
-        public SauceSession(ISauceRemoteDriver driver)
+        public SauceSession()
         {
-            Driver = driver;
             Options = new SauceOptions();
         }
 
         public SauceSession(SauceOptions options)
         {
             Options = options;
-            Driver = new SauceDriver();
         }
 
-        public SauceSession(SauceOptions options, ISauceRemoteDriver driver)
+        public SauceSession(SauceOptions options, IWebDriver driver)
```

Comment:
Session is responsible for creating a driver session, not be initialized with a driver instance.

Reply:
Agreed. The Java case of `SauceOptions` doesn't allow any form of `WebDriver` to be passed in to constructors or getters. This should be the same in the .NET case.